### PR TITLE
feat: apply new color design for search field - ACC-170

### PR DESCRIPTION
**Pull request by BrzozowskiPawel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-170" title="ACC-170" target="_blank">SubtaskACC-170</a>  [iOS] Apply new color design for SearchField
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

In this PR "searchFields" have been updated to conform to the norms required by the contrast accessibility.

### Issues

I have used the reveal app and also the accessibility inspector to check what are the issues with the contrast and than fix it.

### Causes (Optional)

Mostly the color contrast of the text and background. 

### Solutions

Use semantic color for the fix to make text color dependent on the used theme.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift [hunk 1]
```diff
@@ -26,6 +26,45 @@ protocol GiphySearchViewControllerDelegate: AnyObject {
     func giphySearchViewController(_ giphySearchViewController: GiphySearchViewController, didSelectImageData imageData: Data, searchTerm: String)
 }
 
+extension UIImage {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift:29]: I think we don't need this

File: Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift [hunk 2]
```diff
@@ -26,6 +26,45 @@ protocol GiphySearchViewControllerDelegate: AnyObject {
...
+        context?.clip(to: rect, mask: self.cgImage!)
+        context?.fill(rect)
+
+        let newImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        return newImage!
+    }
+}
+
+extension UISearchBar {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift:49]: I think we don't need this

File: Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift [hunk 3]
```diff
@@ -75,6 +74,7 @@ final class SearchHeaderViewController: UIViewController {
         tokenField.tokenSelectedTitleColor = UIColor.from(scheme: .textForeground, variant: colorSchemeVariant)
         tokenField.clipsToBounds = true
         tokenField.textView.placeholderTextColor = UIColor.from(scheme: .tokenFieldTextPlaceHolder, variant: colorSchemeVariant)
+        // TMP COLOR CHANGE: UIColor(red: 0.30, green: 0.34, blue: 0.21, alpha: 1.00) but it makes diffrent UI
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/StartUI/StartUI/SearchHeaderViewController.swift:77]: What does it mean?

File: WireCommonComponents/SemanticColors.swift [hunk 4]
```diff
@@ -25,6 +25,7 @@ enum SemanticColors {
     static let backgroundSwitchOnStateDisabled = UIColor(light: Asset.green300Light, dark: Asset.green300Light)
     static let backgroundSwitchOffStateDisabled = UIColor(light: Asset.gray50, dark: Asset.gray50)
 
+    static let tmpSearchBartextColor = UIColor(light: .black, dark: .white)
```
Comment (KaterinaWire) [inline WireCommonComponents/SemanticColors.swift:28]: We should have another name

File: WireCommonComponents/SemanticColors.swift [hunk 5]
```diff
@@ -43,5 +44,20 @@ private extension UIColor {
             }
         }
     }
+    
+    convenience init(light: UIColor, dark: UIColor) {
```
[the remark this answers is not in the document]
Comment (BrzozowskiPawel) [reply WireCommonComponents/SemanticColors.swift:48]: image
In the develop branch there is an initializer capable of creating an instance with ColorAsset class and for this case, we just needed a SemanticColor of basic black and white.
That's why I have decided it's the best idea to create a nice-looking extra initializer.

File: Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift:29]: Don't forget to delete it

File: WireCommonComponents/SemanticColors.swift [hunk 6]
```diff
@@ -43,5 +44,20 @@ private extension UIColor {
...
+        if #available(iOS 13.0, *) {
+            self.init { traits in
+                return traits.userInterfaceStyle == .dark ? dark : light
+            }
+        } else {
+            switch ColorScheme.default.variant {
```
Comment (KaterinaWire) [inline WireCommonComponents/SemanticColors.swift:54]: Try to replace switch with this:
self.init.from(scheme: .iconSelected, variant: ColorScheme.default.variant)

File: WireCommonComponents/SemanticColors.swift [hunk 7]
```diff
@@ -25,6 +25,7 @@ enum SemanticColors {
     static let backgroundSwitchOnStateDisabled = UIColor(light: Asset.green300Light, dark: Asset.green300Light)
     static let backgroundSwitchOffStateDisabled = UIColor(light: Asset.gray50, dark: Asset.gray50)
 
+    static let defaultSearchBarTextColor = UIColor(light: .black, dark: .white)
```
Comment (KaterinaWire) [inline WireCommonComponents/SemanticColors.swift:28]: I think we can create some static property for black and white semantic color. I don't think we should create an initializer for just one black and white case.

File: Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift (hunk 2 above)
Comment (agisilaos) [reply Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift:29]: @BrzozowskiPawel I don't think that this has been resolved? Can you please delete it?

File: WireCommonComponents/SemanticColors.swift [hunk 8]
```diff
@@ -24,7 +24,7 @@ enum SemanticColors {
     static let backgroundSwitchOffStateEnabled = UIColor(light: Asset.gray70, dark: Asset.gray70)
     static let backgroundSwitchOnStateDisabled = UIColor(light: Asset.green300Light, dark: Asset.green300Light)
     static let backgroundSwitchOffStateDisabled = UIColor(light: Asset.gray50, dark: Asset.gray50)
-
+    static let defaultSearchBarTextColor = UIColor(light: Asset.black, dark: Asset.white)
```
Comment (agisilaos) [inline WireCommonComponents/SemanticColors.swift:27]: Is this referring to the placeholder or the user's input?

**Review by agisilaos — CHANGES_REQUESTED** [review]
Let's take care of Katerina's comments first. Looking good. I left some comments.

File: WireCommonComponents/SemanticColors.swift (hunk 8 above)
Comment (BrzozowskiPawel) [reply WireCommonComponents/SemanticColors.swift:27]: @agisilaos User input.

**Review by agisilaos — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift (hunk 2 above)
Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift:29]: I've unresolved this, @BrzozowskiPawel please address this when you can.

**Review by johnxnguyen — COMMENTED** [review]
I still see an unresolved conversation from Katerina, what is the status on that? Do we need to delete that code or not?

File: Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift (hunk 2 above)
Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/UserInterface/Components/Giphy/GiphySearchViewController.swift:29]: @agisilaos @johnxnguyen I see it but the UISearchBar extension is necessary I guess. It was about the "func imageWithColor()" as I understood and it's gone.

**Pull request merged**